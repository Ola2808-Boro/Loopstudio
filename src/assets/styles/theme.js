export const theme = {
  colors: {
    white: "#ffffff",
    black: "	#000000",
    red: "#FF0000",
    darkGray: "	#8c8c8c",
    veryDarkGray: "#696969",
    vividMulberry: "#C505EB",
    lavenderIndigo: "#8647F8",
  },
  fontSizes: {
    s: "15px",
    m: "20px",
    l: "25px",
    xl: "30px",
  },
};
