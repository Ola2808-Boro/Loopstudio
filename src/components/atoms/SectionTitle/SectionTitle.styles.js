import styled from "styled-components";
import { device } from "helpers/breakpoints";

export const StyledSectionTitle = styled.h2`
  color: ${(props) =>
    props.color ? props.color : ({ theme }) => theme.colors.black};
  font-family: "Josefin Sans", sans-serif;
  font-size: ${({ theme }) => theme.fontSizes.l};
  letter-spacing: 2px;
  line-height: 1.4em;
  text-align: center;
  text-transform: uppercase;
  margin: 20px 0px;
  ${(props) => {
    if (props.isMargin) {
      return `
        padding-bottom: 30px;
        `;
    }
  }}

  @media ${device.desktop} {
    text-align: left;
  }
`;
