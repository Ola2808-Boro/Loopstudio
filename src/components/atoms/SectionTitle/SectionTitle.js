import React from "react";
import PropTypes from "prop-types";
import { StyledSectionTitle } from "components/atoms/SectionTitle/SectionTitle.styles";
const SectionTitle = ({ sectionTitle = "Section Title", color }) => {
  return <StyledSectionTitle color={color}>{sectionTitle}</StyledSectionTitle>;
};

SectionTitle.propTypes = {
  /**Text */
  sectionTitle: PropTypes.string.isRequired,
  /**Color of section title */
  color: PropTypes.string.isRequired,
};

export default SectionTitle;
