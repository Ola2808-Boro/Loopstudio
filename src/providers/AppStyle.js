import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "assets/styles/theme";
import GlobalStyle from "assets/styles/GlobalStyle";
import "assets/styles/fonts.css";

const AppStyle = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};

export default AppStyle;
