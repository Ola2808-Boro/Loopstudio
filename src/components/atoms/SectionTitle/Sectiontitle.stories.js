import SectionTitle from "./SectionTitle";
import { stories } from "helpers/storiesName";

export default {
  title: `${stories.atoms}SectionTitle`,
  component: SectionTitle,
};

const Template = (args) => <SectionTitle {...args} />;

export const Default = Template.bind({});

Default.args = {
  details: "black",
  sectionTitle: "Title",
};
