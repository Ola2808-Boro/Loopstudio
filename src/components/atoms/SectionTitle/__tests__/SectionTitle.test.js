import "jest-styled-components";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import AppStyle from "providers/AppStyle";
import { StyledSectionTitle } from "components/atoms/SectionTitle/SectionTitle.styles";
describe("Test button component", () => {
  it("default style", () => {
    const tree = renderer
      .create(
        <AppStyle>
          <StyledSectionTitle />
        </AppStyle>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
