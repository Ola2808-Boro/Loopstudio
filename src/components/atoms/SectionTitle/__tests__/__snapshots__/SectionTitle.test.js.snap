// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test button component default style 1`] = `
<h2
  className="hdXgWE"
/>
`;
